### DISCUSSION TITLE
Space for labels on right side of sankey plot

### DISCUSSION THREAD

**Discussion by simonpalmer** [HanSolo/charts#89] [Q&A]
Hi, I have the below beautiful Sankey Plot (thank you!). I have been looking through code and docs and cannot see any way that I can prescribe how the labels are treated, especially for the right hand side of the plot. Ideally they would all fall to the right of their respective items and the plot itself would be shrunk to accommodate the longest right-most label. 

I'd love a hint on how to go about this as well as controlling the fonts for plots in general
TIA and I love the library

<img width="1405" alt="Screen Shot 2022-09-14 at 5 42 00 PM" src="https://user-images.githubusercontent.com/749494/190267900-ca13d5a8-bc72-4e2f-a36b-aefbc2ef3eaf.png">

**Comment by HanSolo** [comment]
Glad you like it, well I think I’ve did not spend a lot of time on fonts and labels here. So I will try to find some time to find a solution for that.

**Reply by HanSolo** [reply]
So I've had some time before my flight to Munich and added not a solution for the text on the left and right side of the SankeyPlot but I've added the ability to set a custom font (which still will be autoscaled), make use of the PlotItems text color and set a vertical text position for each PlotItem. At least one can prevent overlapping text in more cases now. I also thought about automatically setting the vertical text position for the very right items if they collide with the ones from the column before. You can find the code when you checkout the latest version of the jdk17 branch.

**Reply by HanSolo** [reply]
Now you can also setAutoAdjustVerticalTextPosition(true) on the SankeyPlot and it will try to avoid text collisions in the last column.

**Reply by simonpalmer** [reply]
This is great, thanks, and I hope you have safe travels. I grabbed the code and had a quick look, but need to focus on it a bit more. Now you have FontMetrix I am looking for a place where that can be incorporated into the item bounds. If I find something sensible I may suggest an edit. I really appreciate that you took a look, thanks.

**Reply by HanSolo** [reply]
Changing the complete chart dependent on the text width is mich more work, so I might look into that in the next days but cannot promise…
